app/controllers/application_controller.rb

```diff
@@ -37,7 +37,7 @@ def switch_locale(&action)
     # if we can't find that, use browser locale
     if locale.nil?
       # the user might their locale set in the browser
-      locale = http_accept_language.compatible_language_from(I18n.available_locales)
+      locale = http_accept_language.preferred_language_from(I18n.available_locales)
```

Comment:
According to the docs, should we actually be using `language_region_compatible_from` instead of `preferred_language_from`?

Reply:
It seems like that form the description, but from a bit of testing I think `preferred_language_from` is better. See this case:

```
Available locales: [:en, :"en-GB", :es, :fr, :"fr-CA", :pt]
User's preferred languages (in order): ["en-AU", "en-GB", "en-US", "en"]
preferred_language_from: en-GB
compatible_language_from: en
language_region_compatible_from: en
```
The user prefers `en-GB` over `en`, but `language_region_compatible_from` chooses `en` anyways because their first preference, `en-AU`, isn't available.

Reply:
Fascinating.  I agree with your approach, then.